tests/lib/nested.sh

```diff
@@ -23,41 +23,86 @@
 : "${NESTED_DISK_LOGICAL_BLOCK_SIZE:=512}"
 
 nested_wait_for_ssh() {
-    # TODO:UC20: the retry count should be lowered to something more reasonable.
-    local retry=800
-    local wait=1
+    local retry=${1:-800}
+    local wait=${2:-1}
 
-    until remote.exec "true"; do
+    until remote.exec "true" &>/dev/null; do
         retry=$(( retry - 1 ))
```

Comment:
nitpick: Is this not a try more than a retry? I think moving `retry=$(( retry - 1 ))` after the `if` and before the `sleep` makes the behaviour more accurate, and possible to give retry == 0

Same for other wait for no variant of the same kind of logic.

Reply:
done